### PR TITLE
[expo-notifications] New monthly scheduled trigger type

### PR DESCRIPTION
**Pull request by douglowder**
# Why

As discussed in #30577 , customers would like a monthly trigger input in addition to the existing daily, weekly, and yearly inputs.

# How

Following the recent changes in scheduled notifications, added a new monthly trigger type using code similar to that used for yearly trigger types.

# Test Plan

- New unit test for the new trigger type
- CI should pass
- Test code added to our app for manual testing

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by douglowder** [comment]
* **#31823** <a href="https://app.graphite.dev/github/pr/expo/expo/31823?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @douglowder and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/triggers/MonthlyTrigger.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
+package expo.modules.notifications.notifications.triggers;
```
Comment (vonovak) [inline packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/triggers/MonthlyTrigger.java:1]: shall we write new code in Kotlin instead?

File: packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/scheduling/NotificationScheduler.kt [hunk 2]
```diff
@@ -197,6 +198,23 @@ open class NotificationScheduler : Module() {
         )
       }
 
+      "monthly" -> {
+        val day = params["day"] as? Number
+        val hour = params["hour"] as? Number
+        val minute = params["minute"] as? Number
+
+        if (day == null || hour == null || minute == null) {
+          throw InvalidArgumentException("Invalid value(s) provided for yearly trigger.")
```
Comment (vonovak) [inline packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/scheduling/NotificationScheduler.kt:207]: the error message could contain the provided values for better debuggability

File: packages/expo-notifications/src/__tests__/Notifications-test.ts [hunk 3]
```diff
@@ -123,6 +123,47 @@ it(`verifies weekly trigger input validation`, async () => {
...
+    day: 32,
+    hour: 12,
+    minute: 30,
+  };
+  const input = {
+    ...notificationTriggerInputTest,
+    trigger,
+  };
+  try {
+    await scheduleNotificationAsync(input);
```
Comment (vonovak) [inline packages/expo-notifications/src/__tests__/Notifications-test.ts:159]: we expect this to throw. However, if it doesn't throw, the test will pass when instead it should fail.

Jest has this https://jestjs.io/docs/expect#expecthasassertions / https://jestjs.io/docs/expect#expectassertionsnumber

But we probably don't have these here. Probably we should throw some error after `await scheduleNotificationAsync(input);` which will be caught by the catch block but the expectations will fail.

At least one more test case here makes the same assumption.

**Review by vonovak — CHANGES_REQUESTED** [review]

File: packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/scheduling/NotificationScheduler.kt (hunk 2 above)
Comment (douglowder) [reply packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/scheduling/NotificationScheduler.kt:207]: Let's review all these error messages and update them in a separate PR.

File: packages/expo-notifications/src/__tests__/Notifications-test.ts (hunk 3 above)
Comment (douglowder) [reply packages/expo-notifications/src/__tests__/Notifications-test.ts:159]: Good point! I'll go through all these tests to make sure that all the tests for exceptions will fail if they do not throw.

Comment (douglowder) [reply packages/expo-notifications/src/__tests__/Notifications-test.ts:163]: Fixed!

File: packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/triggers/MonthlyTrigger.java (hunk 1 above)
Comment (vonovak) [reply packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/triggers/MonthlyTrigger.java:1]: ok, let's do that here, though the interop between Kotlin and Java is pretty seamless so I'd advocate for writing future code in Kotlin only.

**Review by vonovak — APPROVED** [review]

**Review by vonovak — APPROVED** [review]

**Pull request merged**